brainiak/utils/fmrisim.py

```diff
@@ -1595,6 +1603,57 @@ def _generate_noise_temporal_drift(trs,
         phase = (timepoints / (trs - 1) * cycles * 2 * np.pi) + phaseshift
         noise_drift = np.sin(phase)
 
+    elif basis == 'cos_power_drop':
+
+        # Specify when each
+        timepoints = np.linspace(0, trs - 1, trs) * tr_duration
+
+        # Specify the other timing information
+        duration = trs * tr_duration
+
+        # How bases do you have
+        basis_funcs = int(np.floor(2 * duration))
+
+        noise_drift = np.zeros((timepoints.shape[0], basis_funcs))
+        for basis_counter in list(range(1, basis_funcs + 1)):
+            # What steps do you want to take for this basis function
+            random_phase = np.random.rand() * np.pi * 2
+
+            timepoint_phase = (timepoints / duration * np.pi * basis_counter)
+
+            # In radians, what is the value for each time point
+            timepoints_basis = timepoint_phase + random_phase
+
+            # Store the drift from this basis func
+            noise_drift[:, basis_counter - 1] = np.cos(timepoints_basis)
+
+        # Function to return the drop rate for the power of basis functions
+        def power_drop(r, L, F):
```

Comment:
What does this each line do? It may help to explain why th 4 is there, and why the power can both a number without unit ( 1 - r ** (4 * L / F) ) and a number with unit (of time): ( denominator = 1 - r ** (4 * L) )

Reply:
Yeah this part was tricky, Chris did the heavy lifting with the math. Hopefully I have explained it better now.

Reply:
Hi @CameronTEllis Thanks for the explanation below!
Sorry I am still a bit confused so I will try to express my understanding with my new comments at other places of the code.